src/main/clojure/cemerick/pomegranate/aether.clj

```diff
@@ -598,6 +598,9 @@ kwarg to the repository kwarg.
   [& {:keys [repositories coordinates files retrieve local-repo
              transfer-listener offline? proxy mirrors repository-session-fn]
       :or {retrieve true}}]
+  (when repositories
+    (assert (seq repositories)
+            "Empty but truthy `repositories` value found. Please set to nil for using the default, or add a non-empty coll of repositories."))
   (let [repositories (or repositories maven-central)
```

Comment:
I think I'd prefer
`(or (seq repositories) maven-central)`
rather than throwing via `assert`?

Reply:
As of `master`, if one passed `[]` as the repositories one would get an opaque error (`DependencyResolutionException`).

So an assert merely fails faster i.e. it's not a breaking change.

Whereas accepting `[]` is sort of a breaking change since badly configured consumers will start fetching some, but possibly not all stuff (most notably Clojars artifacts!).

Reply:
I might be slow now but what I was suggesting was changing line 604 to 
```clj
(let [repositories (or (seq repositories) maven-central))
```
But perhaps you're saying that `repositories` is used outside of this let-binding and that that leads to the `DependencyResolutionException`?

Reply:
> But perhaps you're saying that 

nope, I understand but disagree with the suggestion.

If this branch succeeds: `(or (seq repositories)` it means that the consumer was badly setup. I don't think pomegranate should swallow that and default to something useful-looking.

That default might be somewhat (or very) different from what the consumer really wants, from Clojars to custom repos.